src/burnchains/tests/db.rs

```diff
@@ -579,20 +591,38 @@ fn test_get_commit_at() {
     // fork off the last stored commit block
     let fork_hdr = BurnchainHeaderHash([90 as u8; 32]);
     let fork_block_header = BurnchainBlockHeader {
-        block_height: 4,
+        block_height: 5,
         block_hash: fork_hdr,
-        parent_block_hash: BurnchainHeaderHash([5 as u8; 32]),
-        num_txs: 0,
+        parent_block_hash: BurnchainHeaderHash([4 as u8; 32]),
+        num_txs: 1,
         timestamp: 4 as u64,
     };
 
+    let mut fork_cmt = cmts[4].clone();
+    fork_cmt.burn_header_hash = fork_hdr.clone();
+    fork_cmt.vtxindex += 1;
+
+    let mut fork_headers = headers.clone();
+    fork_headers[5] = fork_block_header.clone();
+
     burnchain_db
-        .store_new_burnchain_block_ops_unchecked(&burnchain, &headers, &fork_block_header, &vec![])
+        .store_new_burnchain_block_ops_unchecked(
+            &burnchain,
+            &fork_headers,
+            &fork_block_header,
+            &vec![BlockstackOperationType::LeaderBlockCommit(fork_cmt.clone())],
+        )
         .unwrap();
-    headers[4] = fork_block_header;
 
-    let cmt = BurnchainDB::get_commit_at(&burnchain_db.conn(), &headers, 4, 0).unwrap();
-    assert!(cmt.is_none());
+    let cmt = BurnchainDB::get_commit_at(&burnchain_db.conn(), &headers, 5, 0)
+        .unwrap()
+        .unwrap();
+    assert_eq!(cmt, cmts[4]);
+
+    let cmt = BurnchainDB::get_commit_at(&burnchain_db.conn(), &fork_headers, 5, 1)
+        .unwrap()
+        .unwrap();
+    assert_eq!(cmt, fork_cmt);
```

Comment:
Can you add a brief comment here that this test covers #3520 and a description of what it's checking?

Reply:
Better: I added a distinct test for it

Reply:
The test: https://github.com/stacks-network/stacks-blockchain/pull/3524/files#diff-53d26ed8ff095df944d5c90e1c20a7792aaa5032d510a1ffc93bd584e0f58aa5R833